factory/glideFactoryEntry.py

```diff
@@ -1292,7 +1292,7 @@ def unit_work_v3(
             try:
                 entry.log.info("frontend_scitoken supplied, writing to %s" % scitoken_file)
                 chmod(tmpnm, 0o600)
-                os.write(fd, scitoken_data.encode("utf-8"))
+                os.write(fd, (scitoken_data + os.linesep).encode("utf-8"))
                 os.close(fd)
                 util.file_tmp2final(scitoken_file, tmpnm)
```

Comment:
Had a bunch of comments but clicked the wrong button and loast all. Mostly about pre-existing code but I think there is alot of back and forth that can be spared:
1. Maybe the token data should be written as received from the decrypted classad w/o many changes, and a second string used locally for checks, maybe adding the strip() in `token_util.token_str_expired`
2. if stripping the end do also the beginnig. I don't think tokens expect spaces/newlines at the beginning, `scitoken_data.strip("\n ")` . See 1 about doing it in the verify
3. What about instead of os.write, os.close:
```
with fdopen(fd, 'w', encoding="utf-8") as f:
    f.write(f"{scitoken_data}\n")
```
or `print(scitoken_data, file=f)`, both use universal newline (whatever the os wants except when forced in the file opening. close is automatic (also for fd), this defaults to text while mkstemp defaults to binary